File: app/src/main/java/com/better/alarm/interfaces/Alarm.java
package com.better.alarm.interfaces;

import android.net.Uri;

import com.better.alarm.model.Alarmtone;

import java.util.Calendar;

public interface Alarm {
    void enable(boolean enable);

    void snooze();

    void snooze(int hourOfDay, int minute);

    void dismiss();

    void requestSkip();

    boolean isSkipping();

    void delete();

    AlarmEditor edit();

    boolean isSilent();

    Uri getAlert();

    int getId();

    String getLabelOrDefault();

    Alarmtone getAlarmtone();


    /**
     * @return
     * @deprecated
     */
    @Deprecated
    public Calendar getSnoozedTime();
}

File: app/src/main/java/com/better/alarm/interfaces/Alarm.kt
package com.better.alarm.interfaces

import com.better.alarm.model.AlarmValue
import com.better.alarm.model.Alarmtone

/**
 * Alarm 인터스페이스가 어떻게 구성되어있는지
 */
interface Alarm {


    /* enable alarm */
    fun enable(enable: Boolean)
    fun snooze()
    fun snooze(hourOfDay: Int, minute: Int)
    fun dismiss() // 알람 끄기
    fun requestSkip()
    fun isSkipping(): Boolean
    fun delete()

    /** Change something and commit */
    fun edit(func: AlarmValue.() -> AlarmValue)
    val id: Int
    val labelOrDefault: String
    val alarmtone: Alarmtone
    val data: AlarmValue
}

//val id: Int,
//val description: String,
//val choices: List<String>,
//val correctAnswer: Int
